src/model/window-guardian.ts

```diff
@@ -74,18 +74,21 @@ export const makeGuardianBrowserCAPI = (CAPI: CAPIType): CAPIBrowserType => {
 		blockElementType: CAPIElement['_type'],
 		indexName: string,
 	): T[] => {
-		return blocks[0].elements.reduce(
-			(acc: T[], element: CAPIElement, index: number) => {
-				if (element._type === blockElementType) {
-					acc.push({
-						...element,
-						[indexName]: index,
-					} as T);
-				}
-				return acc;
-			},
-			[] as T[],
-		);
+		if (blocks.length > 0) {
+			return blocks[0].elements.reduce(
+				(acc: T[], element: CAPIElement, index: number) => {
+					if (element._type === blockElementType) {
+						acc.push({
+							...element,
+							[indexName]: index,
+						} as T);
+					}
+					return acc;
+				},
+				[] as T[],
+			);
+		}
+		return [];
```

Comment:
👍 Minor style thing - I tend to prefer early return for the edge/error case rather than for the happy case (and I think this is fairly common). So this would instead be:

```
if(blocks.length === 0) return [];

return blocks[0].elements...
```